blogpost-drools-model-compiler.md

```diff
@@ -0,0 +1,116 @@
+the team will ask you to create such reproducer. There are two ways to do it:
+
+1) If you're familiar with the [Drools GitHub repository](https://github.com/kiegroup/drools) you can write the test directly in your own fork of the original repository and create a PR against it. 
+   This is probably the best way to proceed, as it allows all the Drools' developers to check the problem faster
+2) Create another separate project that shows the problem
+
+## Method #1: create a reproducer in drools.git
+
+Start by building the Drools project reading the [Drools Source Code](https://drools.org/code/sourceCode.html) page.
+
+You can either decide to build it using [droolsjbpm-build-bootstrap](https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/master/README.md) or building only the [Drools](https://github.com/kiegroup/drools) module with `mvn clean install`. The second one is definitely faster.
+
+Once you have the project up and running you can open it with your preferred IDE and take a look at the tests in the `drools-model-compiler` module, for example `org.drools.modelcompiler.CompilerTest`. 
+
+If you run these tests you'll see they're executed twice, once against the `DRL` mode and the other against the `PatternDSL`. It's important that the tests run in both ways. If you see a difference in the execution, please create a PR. And if you want to fix it on your own, try - we love to see new contributors.
+
+## Method #2: create a self contained project
+
+Let's use the Drools archetype and verify that your small reproducer is working against Drools Legacy. 
+
+Start by creating a KJAR using the [Maven Archetype](https://mvnrepository.com/artifact/org.kie/kie-drools-archetype), modify the rules and the test to verify everything is working accordingly. The default archetype will run the test against the DRL mode.
+
+Change the generated model to build an executable model KJAR. To do so, switch in the `pom.xml` from `drools-engine-classic` to `drools-engine`. Also add the `drools-model-compiler` dependency.
+
+Compile the project using maven and the command line. Use 
+
+`mvn clean install -DskipTests=true` 
+
+as it'll try run the tests using the classic engine but we don't have the `drools-mvel` dependency in the class path anymore.
+
+![kjar](kjar.gif)
+
+Verify the Executable Model has been built in the KJAR, you can for example using the command
+
+`jar -tf target/name_of_the_kjar.jar`
+
+And you'll see all the Executable Model classes under and the `drools-model` file. 
+You can also check the `target` subfolder of maven and you'll see the generated files.
+Yet another way to do it is to check the maven log for this phrase:
+
+```
+
+INFO] Found 7 generated files in Canonical Model
+[INFO] Generating /Users/lmolteni/git/contribute/reproducer-kjar/target/generated-sources/drools-model-compiler/main/java/./org/example/P41/LambdaExtractor41A2683D222972683028514525A5437B.java
```

Comment:
Is this message still the same after latest PR from Mario to `kie-maven-plugin`?

Reply:
@danielezonca yes the message is the same, but it doesn't actually write the file on the file system, unless you didn't explicitly ask for that by providing the dump folder.